usr/src/cmd/zdb/zdb.c

```diff
@@ -2142,6 +2175,117 @@ dump_label_uberblocks(vdev_label_t *lbl, uint64_t ashift)
 }
 
 static int
+dump_specific_path(objset_t *os, uint64_t obj)
+{
+	int print_header = 1;
+
+	dump_object(os, obj, dump_opt['v'], &print_header);
+
+	return (0);
+}
+
+static int
+dump_specific_dir(objset_t *os, uint64_t obj, char *topname, char *name)
+{
+	int err;
+	uint64_t child_obj;
+	char *slash = strchr(name, '/');
+	char *namep = name;
+	dmu_buf_t *db;
+	dmu_object_info_t doi;
+
+	if (slash != NULL)
+		*slash = '\0';
+	err = zap_lookup(os, obj, namep, 8, 1, &child_obj);
+	if (slash != NULL)
+		*slash = '/';
```

Comment:
We should move this further below so that when we print the error or the verbose info we actually print the path for the directory we are traversing, not the whole path.
e.g.:
if our path is `/testpool/ds1/path/to/our/file`
and `/testpool/ds1/path/to/our` doesn't exist, we'd print debug for:
`/testpool/ds1`
`/testpool/ds1/path`
`/testpool/ds1/path/to`
And then print:
`failed to lookup: /testpool/ds1/path/to/our`